### PR TITLE
CARTO: Implement raster rendering

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/7828

#### Change List
- Support new `MAP_TYPE.RASTER` in `CartoLayer`
- Add `CartoRasterTileLoader` for loading and parsing raster data
- Add `RasterLayer` for rendering data (basically a modified `ColumnLayer` with custom vertex shader)
- Add `RasterTileLayer` to provide tiling support and access to properties in standard format
- Tests (still needs a bit of work)
- Update docs

### CODE REVIEW

File: examples/get-started/pure-js/carto/app.js [hunk 1]
```diff
@@ -1,20 +1,48 @@
 import maplibregl from 'maplibre-gl';
 import {Deck} from '@deck.gl/core';
-import {fetchMap} from '@deck.gl/carto';
+import {CartoLayer, setDefaultCredentials, BASEMAP, MAP_TYPES} from '@deck.gl/carto';
```
Comment (felixpalmer) [inline examples/get-started/pure-js/carto/app.js:3]: Ignore changes to this file, they will be reverted

File: examples/get-started/pure-js/carto/app.js [hunk 2]
```diff
@@ -1,20 +1,47 @@
...
       map.jumpTo({center: [longitude, latitude], ...rest});
     }
-  });
+  }),
+
+  layers: [
+    new CartoLayer({
+      connection: 'deb-bigquery',
+      type: MAP_TYPES.RASTER,
+      data: 'cartodb-data-engineering-team.jarroyo_raster.hillshade1x1_quadbin_2',
```
Comment (alasarr) [inline examples/get-started/pure-js/carto/app.js:31]: This dataset is not public, we need to move it cartobq.testtables.

Comment (alasarr) [reply examples/get-started/pure-js/carto/app.js:31]: Also, we need to make sure this raster has the resolution metadata

File: docs/api-reference/carto/carto-layer.md [hunk 3]
```diff
@@ -69,6 +69,26 @@ function App({viewState}) {
...
+```js
+import DeckGL from '@deck.gl/react';
+import {CartoLayer, MAP_TYPES} from '@deck.gl/carto';
+
+function App({viewState}) {
+  const layer = new CartoLayer({
+    type: MAP_TYPES.RASTER,
+    connection: 'bigquery',
+    data: 'cartobq.testtables.raster',
+    getFillColor: d => [d.properties.temperature, 0, 0]
```
Comment (alasarr) [inline docs/api-reference/carto/carto-layer.md:85]: I prefer to have here a real example raster using bands. `cartobq.testtables.raster` doesn't exist, I suggest to upload a temperature raster at `cartobq.deckgl.temperature_raster`

File: modules/carto/src/api/parseMap.ts [hunk 4]
```diff
@@ -217,7 +217,7 @@ function createChannelProps(
     );
   }
 
-  if (heightField) {
```
Comment (alasarr) [inline modules/carto/src/api/parseMap.ts:220]: It's not related with raster, right?

File: docs/api-reference/carto/carto-layer.md (hunk 3 above)
Comment (alasarr) [reply docs/api-reference/carto/carto-layer.md:85]: Better at `cartobq.public_account.temperature_raster`

**Review by alasarr — CHANGES_REQUESTED** [review]

File: examples/get-started/pure-js/carto/app.js (hunk 2 above)
Comment (felixpalmer) [reply examples/get-started/pure-js/carto/app.js:31]: None of this file is to be commited, it's just for testing - I left a comment before

File: modules/carto/src/api/parseMap.ts (hunk 4 above)
Comment (felixpalmer) [reply modules/carto/src/api/parseMap.ts:220]: Rasters do support extrusion (comes with perf hit), but yes this was an issue I just found while developing that is unrelated to this PR (fetchMap doesn't support raster)

File: docs/api-reference/carto/carto-layer.md (hunk 3 above)
Comment (felixpalmer) [reply docs/api-reference/carto/carto-layer.md:85]: Good plan - I will do once the API is deployed so I can test

**Review by alasarr — APPROVED** [review]

**Pull request merged**